base64 file upload blog (#1383)
pankod/refine: documentation/blog/2021-12-22-base64-upload.md

---

> title: Base64 File Upload
> description: Simple base64 file upload example
> slug: base64-upload
> authors: melih
> tags: [refine, base64, file-upload]
> image: https://refine.dev/img/refine_social.png
> hide_table_of_contents: false

Reviewer: ```suggestion
title: How to upload files from your HTML form using Base64 encoding
description: Uploading files using Base64 encoding is a common practice. In this guide, I'm going to show you how to upload files using base64 encoding
slug: how-to-base64-upload
authors: melih
tags: [javascript, how-to, base64, file-upload]
image: https://refine.dev/img/refine_social.png
hide_table_of_contents: false
```

---

import upload from '@site/static/img/blog/2021-12-22-base64-upload/upload.png';
import overview from '@site/static/img/blog/2021-12-22-base64-upload/overview.png';

> In this article, we will make and examine a simple base64 file upload example.

Reviewer: ```suggestion
Uploading files using Base64 encoding is a common practice. In this guide, I'm going to show you how to upload files using base64 encoding
```

<!--truncate-->

> ## What is Base64 ?

Reviewer: ```suggestion
## What is Base64 encoding?
```

            integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC"
            crossorigin="anonymous"
        />
        <title>Base64 File Upload</title>
    </head>

    <body>
        <div style="margin: 24px">
            <h2>Upload Image</h2>
        </div>

        <script src="./index.js"></script>
    </body>
</html>
```

Then we need to use the HTML [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) tag to receive an image file from the user.

We will use `file input` because it must be the input type file we want to receive.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="./index.css" />
    <link
      href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css"
      rel="stylesheet"
      integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC"
      crossorigin="anonymous"
    />

    <title>Base64 File Upload</title>
  </head>
  <body>
    <div style="margin: 24px">
      <h2>Upload Image</h2>
    </div>

>     //highlight-start
>     <div style="margin: 16px; padding: 16px">
>       <input
>         class="form-control form-control-lg"
>         id="selectAvatar"
>         type="file"
>       />
>     //highlight-end
>     </div>

Reviewer: ```suggestion
    //highlight-start
    <div style="margin: 16px; padding: 16px">
      <input
        class="form-control form-control-lg"
        id="selectAvatar"
        type="file"
      />
    </div>
    //highlight-end
```

---
title: How to upload files from your HTML form using Base64 encoding
description: Uploading files using Base64 encoding is a common practice. In this guide, I'm going to show you how to upload files using base64 encoding
slug: how-to-base64-upload
authors: melih

> tags: [javascript, how-to, base64, file-upload]

Reviewer: ```suggestion
tags: [JavaScript, how-to, base64, file-upload]
```

        />

        <title>Base64 File Upload</title>
    </head>
    <body>
        <div style="margin: 24px">
            <h2>Upload Image</h2>
        </div>

        <div style="margin: 16px; padding: 16px">
            <input
                class="form-control form-control-lg"
                id="selectAvatar"
                type="file"
            />
        </div>
        //highlight-start
        <div class="container">
            <div class="row">
                <div class="col">
                    <h6>Image Preview:</h6>
                    <img class="img" id="avatar" />
                </div>
                <div class="col">
                    <h6>Base64 Output</h6>
                    <textarea id="textArea" rows="30" cols="50"></textarea>
                </div>
            </div>
        </div>
        //highlight-end
        <script src="./index.js"></script>
    </body>
</html>
```

<div class="img-container">
    <div class="window">
        <div class="control red"></div>
        <div class="control orange"></div>
        <div class="control green"></div>
    </div>
    <img src={upload} alt="upload_screen" />
</div>
<br />

> Our fields are ready to display image file and Base64 code. Now let's do our operations on the javascript side.

Reviewer: ```suggestion
Our fields are ready to display the image file and Base64 code. Now let's do our operations on the JavaScript side.
```

     style={{width: "100%", height:"80vh", border: "0px", borderRadius: "8px", overflow:"hidden"}}
     title="base64-upload-file"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
></iframe>

<br/>
<br/>

## Are You Looking React Web Framework?
A React-based framework for building data-intensive applications in no time. **refine** offers lots of out-of-the box functionality for rapid development, without compromising extreme customizability. Use-cases include, but are not limited to admin panels, B2B applications and dashboards.

⚙️ Zero-configuration: One-line setup with superplate. It takes less than a minute to start a project.

📦 Out-of-the-box : Routing, networking, authentication, state management, i18n and UI.

🔌 Backend Agnostic : Connects to any custom backend. Built-in support for REST API, Strapi, NestJs CRUD, Airtable, Supabase, Appwrite and Altogic.

> 📝 Native Typescript Core : You can always opt out for plain Javascript.

Reviewer: ```suggestion
📝 Native Typescript Core : You can always opt-out for plain JavaScript.
```
